### PR TITLE
Fix help window dimension and allow scrolling

### LINKED ISSUE #226: The current help window is too big because of the number of commands that we added
The window is too big and hinder the user from adjusting the view and closing it. Ideally this window should have its content scrollable and of fix dimension, so as to avoid being marked as feature flaw

### PR DESCRIPTION
**Pull request by MAOXIONGKAI**
Closes #226

Since fxml does not support percentage size when defining component dimension, the current fix attempt uses fix height of 600 as the way to restrict the height. Though not the best practice but should work for most people, but if anything issue happen we can discuss about how to fix this properly

### CODE REVIEW

**Review by huangjiaxi1111 — COMMENTED** [review]
Screenshot 2024-11-07 at 7 23 19 PM
It looks like this on mac, can the font size be larger?

File: src/main/resources/view/HelpWindow.css [hunk 1]
```diff
@@ -17,3 +17,7 @@
 #helpMessageContainer {
     -fx-background-color: derive(#1d1d1d, 20%);
 }
+
+#helpMessageScrollPane {
+    -fx-max-height: 600
+}
```
Comment (weiseng18) [inline src/main/resources/view/HelpWindow.css:23]:
```css
#helpMessageContainer {
    -fx-background-color: derive(#1d1d1d, 20%);
    -fx-min-width: 1080;
}

#helpMessageScrollPane {
    -fx-max-height: 600;
    -fx-min-width: 1100;
}
```

this seems to work for me, 1080 is calculated by taking 1100-20

**Review by weiseng18 — DISMISSED** [review]
lgtm

File: src/main/resources/view/HelpWindow.fxml [hunk 2]
```diff
@@ -19,6 +20,7 @@
       <stylesheets>
         <URL value="@HelpWindow.css" />
       </stylesheets>
+      <ScrollPane fx:id="helpMessageScrollPane">
```
Comment (huangjiaxi1111) [inline src/main/resources/view/HelpWindow.fxml:23]: Try this <ScrollPane fx:id="helpMessageScrollPane" hbarPolicy="NEVER" fitToWidth="true">

Comment (huangjiaxi1111) [inline src/main/resources/view/HelpWindow.fxml:23]: <ScrollPane fx:id="helpMessageScrollPane" hbarPolicy="NEVER" fitToWidth="true">

Comment (huangjiaxi1111) [inline src/main/resources/view/HelpWindow.fxml:23]: try this `<ScrollPane fx:id="helpMessageScrollPane" hbarPolicy="NEVER" fitToWidth="true">`

Comment (MAOXIONGKAI) [reply src/main/resources/view/HelpWindow.fxml]: Thanks for suggestion! It works very good

File: src/main/resources/view/HelpWindow.css (hunk 1 above)
Comment (MAOXIONGKAI) [reply src/main/resources/view/HelpWindow.css]: Thanks for the help, the window looks nice

**Review by weiseng18 — APPROVED** [review]
lgtm

**Pull request merged**